smart-contracts/cookbook/smart-contracts/deploy-basic/basic-hardhat.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Deploy Basic Contract with Hardhat
+description: Learn how to deploy a basic smart contract to Polkadot Hub using Hardhat, ideal for professional workflows that require comprehensive testing and debugging.
+categories: Smart Contracts
+---
+
+# Deploy Basic Contract with Hardhat
+
+## Introduction
+
+This guide demonstrates how to deploy a basic Solidity smart contract to Polkadot Hub using [Hardhat](https://hardhat.org/){target=\_blank}, which provides a comprehensive development environment with built-in testing, debugging, and deployment capabilities. It's ideal for professional development workflows and team projects.
+
+## Prerequisites
+
+Before you begin, ensure you have the following:
+
+- A basic understanding of [Solidity](https://www.soliditylang.org/){target=\_blank} programming.
+- [Node.js](https://nodejs.org/en/download){target=\_blank} v22.13.1 or later installed.
+- Test tokens for gas fees, available from the [Polkadot faucet](https://faucet.polkadot.io/){target=\_blank}. See [Get Test Tokens](/smart-contracts/faucet/#get-test-tokens){target=\_blank} for a guide to using the faucet.
+- A wallet with a private key for signing transactions.
+
+## Set Up Your Project
+
+Use the following terminal commands to create a directory and initialize your Hardhat project inside of it:
+
+```bash
+mkdir hardhat-deployment
+cd hardhat-deployment
+npx hardhat --init
+```
+
+## Configure Hardhat
+
+Open `hardhat.config.js` and add `polkadotHubTestnet` to the `networks` configuration as highlighted in the following example code:
+
+```javascript title='hardhat.config.js' hl_lines='39-43'
+--8<-- 'code/smart-contracts/cookbook/smart-contracts/deploy-basic-contract/basic-hardhat/hardhat.config.js'
```

Comment:
Looks like a TypeScript file? This line stood out to me specifically:

```
const config: HardhatUserConfig = {
```

Cause it uses the `HardhatUserConfig` type